okio/src/commonMain/kotlin/okio/-Util.kt

```diff
@@ -88,3 +89,28 @@ internal fun arrayRangeEquals(
   }
   return true
 }
+
+internal fun Byte.toHexString(): String {
+  val result = CharArray(4)
+  result[0] = '0'
+  result[1] = 'x'
+  result[2] = HEX_DIGITS[this shr 4 and 0xf]
+  result[3] = HEX_DIGITS[this       and 0xf] // ktlint-disable no-multi-spaces
+  return String(result)
+}
+
+internal fun Int.toHexString(): String {
+  // Should this trim leading 0's?
```

Comment:
I can improve these implementations if we want to trim leading `0`s. I went for a fast and dirty implementation at first.

Reply:
Thumbs up. You need to position the `x` where the last leading `0` is, and call the String constructor that takes an offset and a count.

Reply:
Also unclear whether bytes read nicer as `0x0f` or `0xf`. For ints I think trimming leading zeros is good.

Reply:
Actually I think we probably want to do something weirder for these.

We should delegate! return Buffer().writeUtf8("0x").writeHexadecimalUnsignedLong(this).readUtf8()

Reply:
Sadly `writeHexadecimalUnsignedLong` is still TODO. It requires figuring out how many characters the number is going to be which currently relies on JVM specific methods. I'm hoping to use the same binary tree style approach used by `commonWriteDecimalLong` instead.

Reply:
FWIW, you *might* be able to do numberOfTrailingZeros and highestOneBit yourself pretty easily, there’s a trick in [Hacker’s Delight](https://hackersdelight.org/) or related sites.

http://aggregate.org/MAGIC/

Reply:
https://github.com/square/okio/issues/620